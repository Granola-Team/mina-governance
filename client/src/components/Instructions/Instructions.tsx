import { useKeywordStore } from "@/pages/Keyword/Keyword.store";
import React from "react";
import shallow from "zustand/shallow";

export const Instructions: React.FC = () => {
  const { key } = useKeywordStore((state) => ({ key: state.key }), shallow);

  return (
    <div className="content-full-width px-2 md:px-4 lg:px-8 mt-4 md:mt-6">
      <div className="bg-gray-2 border border-gray-7 rounded-xl w-full">
        <div className="flex flex-col justify-center px-3 md:px-6 py-2 gap-1.5">
          <div className="flex-col flex md:flex-row items-center justify-between">
            <span className="semibold text-lg md:text-xl leading-8">
              Instructions For Testnet
            </span>
          </div>
          <div className="flex flex-col items-start justify-start md:gap-1">
            <p className="font-semibold">
              Step 1 -{" "}
              <span className="text-[0.875rem] font-normal">
                Create your wallet (Auro, Clorio & Staking Power)
              </span>
            </p>

            <p className="font-semibold">
              Step 2 -{" "}
              <span className="text-[0.875rem] font-normal">
                Get testnet MINA from the official{" "}
                <a
                  href="https://faucet.minaprotocol.com/"
                  className="text-OrangeMINA hover:opacity-80 transition-all duration-200"
                >
                  <u>Faucet</u>
                </a>
              </span>
            </p>

            <p className="font-semibold">
              Step 3 -{" "}
              <span className="text-[0.875rem] font-normal">
                To vote, send yourself a transaction with the memo{" "}
                <span className="font-semibold">&apos;{key}&apos;</span>
              </span>
            </p>

            <p>
              <span className="text-[0.875rem] font-normal">
                Your vote should appear in 10-15 minutes
              </span>
            </p>
          </div>
          <div className="flex-col flex md:flex-row items-center justify-between">
            <span className="semibold text-lg md:text-xl leading-8">
              Instructions For Mainnet
            </span>
          </div>
          <div className="flex flex-col items-start justify-start md:gap-1">
            <p className="font-semibold">
              {" "}
              <span className="text-[0.875rem] font-normal">
              Send yourself a transaction with the keyword(s) in the memo field
              <ul>
                To vote in favor of the proposal enter {" "} in the memo field
                To vote against the proposal enter no {" "} in the memo field
                <span className="font-semibold">&apos;{key}&apos;</span>
              </ul>
              </span>
            </p>

            <p className="font-semibold">
              <span className="text-[0.875rem] font-normal">
                Refresh the dashboard - your vote should appear in the dashboard within 10-15 minutes
                <span className="font-semibold">&apos;{key}&apos;</span>
              </span>
            </p>

            <p className="font-semibold">
              <span className="text-[0.875rem] font-normal">
              See Frequently Asked Questions for more information ({" "}), including how to send yourself a transaction
                <a
                  href="https://docs.google.com/document/d/1cMq2QoE_n61QGycsxRhCPirIAWNDXeWf/edit"
                  className="font-semibold hover:opacity-80 transition-all duration-200"
                >
                  FAQs
                </a>
              </span>
            </p>

          </div>
        </div>
      </div>
    </div>
  );
};
