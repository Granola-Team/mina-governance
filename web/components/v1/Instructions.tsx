import { Link, Stack, Typography } from '@mui/material';

import { SectionLayout } from './SectionLayout';

export type InstructionsProps = {
  keyword: string;
};

export const Instructions = ({ keyword }: InstructionsProps) => {
  console.log('Hello');
  return (
    <SectionLayout>
      <Typography fontSize={22} fontWeight={600}>
        Instructions
      </Typography>
      <Stack>
        <Typography variant="subtitle2">Send yourself a transaction with the keyword(s) in the memo field:</Typography>
        <Typography variant="subtitle2" ml={1}>
          · To vote in favor of the proposal, enter in the memo field &apos;{keyword}&apos;
        </Typography>
        <Typography variant="subtitle2" ml={1}>
          · To vote against of the proposal, enter in the memo field &apos;no {keyword}&apos;
        </Typography>
      </Stack>

      <Stack>
        <Typography variant="subtitle2">
          Refresh the dashboard, your vote should appear in the dashboard within 10-15 minutes
        </Typography>
        <Typography variant="subtitle2">
          A vote will first appear in a pending state. After ten blocks have been confirmed the voting status will be
          marked as Canonical where appropriate.
        </Typography>
      </Stack>

      <Stack>
        <Typography variant="subtitle2">
          See Frequently Asked Questions{' '}
          <Link
            href="https://forums.minaprotocol.com/t/on-chain-voting-frequently-asked-questions-faq/5959"
            target="_blank"
            rel="noreferrer"
            color="#FF6739"
          >
            (FAQs)
          </Link>{' '}
          for more information, including how to send yourself a transaction
        </Typography>
        <Typography variant="subtitle2">
          If you find an issue, a bug or simply have a question, please feel free to write us some{' '}
          <Link
            href="https://github.com/Granola-Team/blog/blob/main/voting-results-instructions.md"
            target="_blank"
            rel="noreferrer"
            color="#FF6739"
          >
            feedback
          </Link>
          . We would love to hear your thoughts!
        </Typography>
      </Stack>
    </SectionLayout>
  );
};
