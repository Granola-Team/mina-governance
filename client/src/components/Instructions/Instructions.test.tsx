import { expect } from "vitest";
import { screen, fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Instructions } from "./Instructions";

test("should navigate to faucet when link is clicked", async () => {
    const rendered = render(<Instructions />);
    const link = rendered.getByText("FAQs");
    fireEvent.click(link);
    expect(screen.getByText("FAQs").closest("a"))
        .toHaveAttribute("href", "https://docs.google.com/document/d/1cMq2QoE_n61QGycsxRhCPirIAWNDXeWf/edit");
});

test("Instructions are rendered and visible", async () => {
    render(<Instructions />);
    expect(screen.getByText("Instructions")).toBeInTheDocument();
});

test("Snapshot test passes", async () => {
    const instructions = render(<Instructions />);
    expect(instructions).toMatchSnapshot();
});
