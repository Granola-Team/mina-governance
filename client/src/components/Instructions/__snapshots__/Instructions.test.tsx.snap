// Vitest Snapshot v1

exports[`Snapshot test passes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content-full-width px-2 md:px-4 lg:px-8 mt-4 md:mt-6"
      >
        <div
          class="bg-gray-2 border border-gray-7 rounded-xl w-full"
        >
          <div
            class="flex flex-row justify-between px-3 md:px-6 py-2 gap-1.5"
          >
            <div
              class="flex flex-col items-start justify-start md:gap-1"
            >
              <div
                class="flex-col flex md:flex-row items-center justify-between"
              >
                <span
                  class="semibold text-lg md:text-xl leading-8"
                >
                  Instructions
                </span>
              </div>
              <p
                class="font-semibold"
              >
                 
                <span
                  class="text-[0.875rem] font-normal"
                >
                  Send yourself a transaction with the keyword(s) in the memo field
                  <ul
                    class="list-disc list-inside"
                  >
                    <li>
                      To vote in favor of the proposal, enter in the memo field 
                       
                      <span
                        class="font-semibold"
                      >
                        '
                        '
                      </span>
                    </li>
                    <li>
                      To vote against the proposal enter in the memo field 
                       
                      <span
                        class="font-semibold"
                      >
                         'no 
                        '
                      </span>
                    </li>
                  </ul>
                </span>
              </p>
              <p
                class="font-semibold"
              >
                <span
                  class="text-[0.875rem] font-normal"
                >
                  Refresh the dashboard - your vote should appear in the dashboard within 10-15 minutes
                </span>
              </p>
              <p
                class="font-semibold"
              >
                <span
                  class="text-[0.875rem] font-normal"
                >
                  See Frequently Asked Questions (
                  <a
                    class="text-OrangeMINA hover:opacity-80 transition-all duration-200"
                    href="https://faucet.minaprotocol.com/"
                  >
                    <u>
                      FAQs
                    </u>
                  </a>
                  ) for more information, including how to send yourself a transaction
                </span>
              </p>
              <p
                class="font-semibold"
              >
                <span
                  class="text-[0.875rem] font-normal"
                >
                  If you find an issue, a bug or simply have a question, please feel free to write us some 
                  <a
                    class="text-OrangeMINA hover:opacity-80 transition-all duration-200"
                    href="/"
                  >
                    <u>
                      feedback
                    </u>
                  </a>
                  . We would love to hear your thoughts!
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content-full-width px-2 md:px-4 lg:px-8 mt-4 md:mt-6"
    >
      <div
        class="bg-gray-2 border border-gray-7 rounded-xl w-full"
      >
        <div
          class="flex flex-row justify-between px-3 md:px-6 py-2 gap-1.5"
        >
          <div
            class="flex flex-col items-start justify-start md:gap-1"
          >
            <div
              class="flex-col flex md:flex-row items-center justify-between"
            >
              <span
                class="semibold text-lg md:text-xl leading-8"
              >
                Instructions
              </span>
            </div>
            <p
              class="font-semibold"
            >
               
              <span
                class="text-[0.875rem] font-normal"
              >
                Send yourself a transaction with the keyword(s) in the memo field
                <ul
                  class="list-disc list-inside"
                >
                  <li>
                    To vote in favor of the proposal, enter in the memo field 
                     
                    <span
                      class="font-semibold"
                    >
                      '
                      '
                    </span>
                  </li>
                  <li>
                    To vote against the proposal enter in the memo field 
                     
                    <span
                      class="font-semibold"
                    >
                       'no 
                      '
                    </span>
                  </li>
                </ul>
              </span>
            </p>
            <p
              class="font-semibold"
            >
              <span
                class="text-[0.875rem] font-normal"
              >
                Refresh the dashboard - your vote should appear in the dashboard within 10-15 minutes
              </span>
            </p>
            <p
              class="font-semibold"
            >
              <span
                class="text-[0.875rem] font-normal"
              >
                See Frequently Asked Questions (
                <a
                  class="text-OrangeMINA hover:opacity-80 transition-all duration-200"
                  href="https://faucet.minaprotocol.com/"
                >
                  <u>
                    FAQs
                  </u>
                </a>
                ) for more information, including how to send yourself a transaction
              </span>
            </p>
            <p
              class="font-semibold"
            >
              <span
                class="text-[0.875rem] font-normal"
              >
                If you find an issue, a bug or simply have a question, please feel free to write us some 
                <a
                  class="text-OrangeMINA hover:opacity-80 transition-all duration-200"
                  href="/"
                >
                  <u>
                    feedback
                  </u>
                </a>
                . We would love to hear your thoughts!
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
